lib/Share/RoomShareProvider.php

```diff
@@ -155,7 +155,7 @@ public function create(IShare $share): IShare {
 		foreach ($existingShares as $existingShare) {
 			if ($existingShare->getSharedWith() === $share->getSharedWith()) {
 				// FIXME Should be moved away from GenericEvent as soon as OCP\Share20\IManager did move too
-				$this->dispatcher->dispatch(self::EVENT_SHARE_FILE_AGAIN, new GenericEvent($existingShare));
+				$this->dispatcher->dispatch(new GenericEvent($existingShare), self::EVENT_SHARE_FILE_AGAIN);
```

Comment:
@nickvergessen I think this solved a bug.

Reply:
https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/private/EventDispatcher/EventDispatcher.php#L86-L87

I think it explodes :P

Reply:
Crazy.
Psalm point to this file:
https://github.com/symfony/event-dispatcher-contracts/blob/main/EventDispatcherInterface.php#L32

I changed now the property to `OCP\EventDispatcher\IEventDispatcher` and solved the psalm issue.